consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -27,8 +27,10 @@ trait BaseApi extends Closable {
     RetryPolicy.backoff(backoffs) {
       // We will assume 5xx are retryable, everything else is not for now
       case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+      // Looks like it could be either of 2 depending on where FutureCancelledException occurred
```

Comment:
I don't understand this comment.  I think it should be something like `Don't retry interrupted failures` or something like that